test/functional/test_framework/test_framework.py

```diff
@@ -158,6 +161,21 @@ def main(self):
             self.options.tmpdir = tempfile.mkdtemp(prefix=TMPDIR_PREFIX)
         self._start_logging()
 
+        # Seed the PRNG.  Note that testruns are only reproducible if only a
+        # single thread accesses the PRNG (see
+        # https://docs.python.org/3/library/random.html#notes-on-reproducibility).
+        # The network thread shouldn't access random. If we need to change the
+        # network thread to access randomness, it should instantiate its own
+        # random.Random object.
+        if self.options.randseed is not None:
+            self.log.debug("Using user supplied random seed {}".format(self.options.randseed))
+            seed = self.options.randseed
+        else:
+            seed = random.randrange(sys.maxsize)
```

Comment:
Alternatively, to use `self.options.randseed` once and avoid a `not` conditional:
```python3
        seed = self.options.randseed

        if seed is None:
            seed = random.randrange(sys.maxsize)
        else:
            self.log.debug("Using user supplied random seed {}".format(seed))
```
Feel free to ignore, these may be Ruby idioms more than Python ones, as would be doing the assignment in the conditional à la `if seed := self.options.randseed is None:` which IIUC is coming to Python 3.8.

Reply:
Looks good. I've taken your version